aws-apigateway-go-routes/main.go

```diff
@@ -0,0 +1,206 @@
+				},
+				{ // Proxy requests to another service
+					Path: "proxy",
+					Target: apigateway.TargetArgs{
+						Type: apigateway.IntegrationType_Http_proxy,
+						Uri:  pulumi.String("https://www.google.com"),
+					},
+				},
+				{ // Use Swagger to define an HTTP proxy route
+					Path:   "swagger",
+					Method: &getMethod,
+					Data: map[string]interface{}{
+						"x-amazon-apigateway-integration": map[string]interface{}{
+							"httpMethod":          "GET",
+							"passthroughBehavior": "when_no_match",
+							"type":                "http_proxy",
+							"uri":                 "https://httpbin.org/uuid",
+						},
+					},
+				},
+				{ // Authorize requests using Cognito
+					Path:         "cognito-authorized",
+					Method:       &getMethod,
+					EventHandler: helloHandler,
+					Authorizers: []apigateway.AuthorizerArgs{
+						{
+							ParameterName:  "Authorization",
+							IdentitySource: []string{"method.request.header.Authorization"},
+							ProviderARNs:   pulumi.StringArray(pulumi.StringArray{userPool.Arn}),
+						},
+					},
+				},
+				{
+					Path:         "lambda-authorized",
+					Method:       &getMethod,
+					EventHandler: helloHandler,
+					Authorizers: []apigateway.AuthorizerArgs{
+						{
+							AuthType:       &custom,
+							ParameterName:  "Authorization",
+							Type:           &request,
+							IdentitySource: []string{"method.request.header.Authorization"},
+							Handler:        authLambda,
+						},
+					},
+				},
+				{
+					Path:           "key-authorized",
+					Method:         &getMethod,
+					EventHandler:   helloHandler,
+					ApiKeyRequired: &apiKeyRequired,
+				},
+			},
+		})
+		if err != nil {
+			return err
+		}
+
+		// Define whole API using swagger (OpenAPI)
+		swaggerAPI, err := apigateway.NewRestAPI(ctx, "swagger-api", &apigateway.RestAPIArgs{
+			SwaggerString: pulumi.String(`{
+				"swagger": "2.0",
+				"info": {
+					"title": "example",
+					"version": "1.0"
+				},
+				"paths": {
+					"/": {
+						"get": {
+							"x-amazon-apigateway-integration": {
+								"httpMethod": "GET",
+								"passthroughBehavior": "when_no_match",
+								"type": "http_proxy",
+								"uri": "https://httpbin.org/uuid"
+							}
+						}
+					}
+				},
+				"x-amazon-apigateway-binary-media-types": ["*/*"]
+			}`),
+		})
+		if err != nil {
+			return err
+		}
+
+		// Create an API key to manage usage
+		apiKey, err := awsapigateway.NewApiKey(ctx, "api-key", &awsapigateway.ApiKeyArgs{})
+		if err != nil {
+			return err
+		}
+		apiId := restAPI.Api.ApplyT(func(api *awsapigateway.RestApi) pulumi.StringOutput {
+			return api.ID().ToStringOutput()
+		}).ApplyT(func(id interface{}) string {
+			return id.(string)
+		}).(pulumi.StringOutput)
+		stageName := restAPI.Stage.ApplyT(func(stage *awsapigateway.Stage) pulumi.StringOutput {
+			return stage.StageName
+		}).ApplyT(func(stageName interface{}) string {
+			return stageName.(string)
+		}).(pulumi.StringOutput)
```

Comment:
This double-apply looks a bit wild. Aren't we getting the stageName output from the first ApplyT? Same question to the `apiId` above.

Reply:
Yes it does appear to be required. The first apply retrieve the sub-property, which itself is an output needing to be resolved. The second apply is to allow us to convert the unknown value to the correct type.